lib/processors/themeBuilderWorker.js

```diff
@@ -0,0 +1,291 @@
+import workerpool from "workerpool";
+import themeBuilder from "./themeBuilder.js";
+import {createResource} from "@ui5/fs/resourceFactory";
+import {Buffer} from "node:buffer";
+
+/**
+ * Task to build library themes.
+ *
+ * @private
+ * @function default
+ * @static
+ *
+ * @param {object} parameters Parameters
+ * @param {MessagePort} parameters.fsInterfacePort
+ * @param {object[]} parameters.themeResources Input array of Uint8Array transferable objects
+ * 	that are the less sources to build upon. By nature those are @ui5/fs/Resource.
+ * @param {object} parameters.options Less compiler options
+ * @returns {Promise<object[]>} Resulting array of Uint8Array transferable objects
+ */
+export default async function execThemeBuild({
+	fsInterfacePort,
+	themeResources = [],
+	options = {}
+}) {
+	const fsThemeResources = uint8ToResources(themeResources);
+	const fsReader = new FsWorkerThreadInterface(fsInterfacePort);
+
+	const result = await themeBuilder({
+		resources: fsThemeResources,
+		fs: fsReader,
+		options
+	});
+
+	return resourcesToUint8(result);
+}
+
+/**
+ * Casts @ui5/fs/Resource-s into an Uint8Array transferable object
+ *
+ * @param {@ui5/fs/Resource[]} resourceCollection
+ * @returns {Promise<object[]>}
+ */
+export async function resourcesToUint8(resourceCollection) {
+	return Promise.all(
+		resourceCollection.map(async (res) => ({
+			transferable: await res.getBuffer(),
+			path: res.getPath()
+		}))
+	);
+}
+
+/**
+ * Casts Uint8Array into @ui5/fs/Resource-s transferable object
+ *
+ * @param {Promise<object[]>} resources
+ * @returns {@ui5/fs/Resource[]}
+ */
+export function uint8ToResources(resources) {
+	return resources.map((res) => {
+		// res.transferable is an Uint8Array object and needs to be cast
+		// to a Buffer in order to be read correctly.
+		return createResource({path: res.path, buffer: Buffer.from(res.transferable)});
```

Comment:
`transferable` is already a buffer now

```suggestion
		return createResource({path: res.path, buffer: res.transferable));
```

Reply:
Unfortunatelly, this is not correct. The case is that when it goes through the `postMessage` on the other side it is received as `Uint8Array` and that's why we need to explicitly cast it back to `Buffer`